src/idpi/mars.py

```diff
@@ -93,21 +95,35 @@ def dump(self):
             exclude_none=True,
         )
 
-    def to_fdb(self):
+    def _param_id(self):
         mapping = _load_mapping()
-        param_id = mapping[self.param]["cosmo"]["paramId"]
-        staggered = mapping[self.param]["cosmo"].get("vertStag", False)
+        if isinstance(self.param, Iterable) and not isinstance(self.param, str):
+            return [mapping[param]["cosmo"]["paramId"] for param in self.param]
+        return mapping[self.param]["cosmo"]["paramId"]
 
+    def _staggered(self):
+        mapping = _load_mapping()
+        if isinstance(self.param, Iterable) and not isinstance(self.param, str):
+            first, *others = (
+                mapping[param]["cosmo"].get("vertStag", False) for param in self.param
+            )
+            if not all(first == other for other in others):
+                raise ValueError("Not all fields have the same staggering.")
+            return first
+        return mapping[self.param]["cosmo"].get("vertStag", False)
```

Comment:
I think it would be one miss per non staggered param. This is however a good point, I had forgotten that FDB does not raise on misses. So should I change the logic to make levelist the staggered one if any of the params is staggered? I also have little clue of the performance impact that that could have. We could still segregate staggered from non-staggered elsewhere in the code if that turns out to be a big performance hit.

Reply:
I would probably opt for if >1 param is staggered, combine the requests and set all to have the levelist of the staggered param, until we see a considerable performance impact, which might not. From what I heard from the FDB team it should be ok if the request is still mostly 'densely' fulfilled.